common/internal_model/src/main/scala/uk/ac/wellcome/models/parse/FreeformDate.scala

```diff
@@ -0,0 +1,103 @@
+package uk.ac.wellcome.models.parse
+
+import java.time.LocalDate
+
+import uk.ac.wellcome.models.work.internal.InstantRange
+
+/**
+  *  Models freeform text representing some date or period
+  *
+  *  @param date The date or date range
+  *  @param inferred Whether the date is known exactly or inferred
```

Comment:
Unless `inferred` has a specific use case, I would be careful about keeping it around if "it might be useful" is the reason for having it as I find it makes understanding what the code _is for_ harder.